### PR TITLE
Use jrpc server to set injectors

### PR DESCRIPTION
**Pull request by AsterNighT**

### CODE REVIEW

**Comment by AsterNighT** [comment]
I work on the original branch so pls squash the commits when merging

**Comment by YangKeao** [comment]
> I work on the original branch so pls squash the commits when merging

You could cherry-pick these commits to another branch which checkouts from current master. There won't be any conflict (I guass).

**Comment by AsterNighT** [comment]
@Hexilee @YangKeao PTAL

**Comment by AsterNighT** [comment]
> > I work on the original branch so pls squash the commits when merging
> 
> You could cherry-pick these commits to another branch which checkouts from current master. There won't be any conflict (I guass).

Sound reasonable, but I've fixed out the conflict this time 😭

File: src/hookfs/mod.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ use nix::unistd::{
 
 use tracing::{debug, error, instrument, trace};
 
-use std::{collections::{HashMap, LinkedList}};
+use std::collections::{HashMap, LinkedList};
```
Comment (Hexilee) [inline src/hookfs/mod.rs]: These imports are chaotic, could you fix them in this PR? We should import them into three groups like following:

```rust
use std::collections::{HashMap, LinkedList};
...(other imports from standard crates)
<blank line>
use tracing::{debug, error, instrument, trace};
...(other imports from third-party crates)
<blank line>
use runtime::spawn_blocking;
...(other imports from current crate)
```

File: src/jsonrpc.rs [hunk 2]
```diff
@@ -1,61 +1,93 @@
...
+    fn update(&self, config: Vec<InjectorConfig>) -> Result<String> {
+        trace!("rpc update called");
+        if let Err(e) = &*self.status.lock().unwrap() {
+            return Ok(e.to_string());
+        }
+        let injectors = MultiInjector::build(config);
+        if let Err(e) = &injectors {
+            return Ok(e.to_string());
         }
+        futures::executor::block_on((async || {
```
Comment (Hexilee) [inline src/jsonrpc.rs:88]: Maybe we need async json-rpc like [jsonrpc-v2](https://github.com/kardeiz/jsonrpc-v2).

**Review by Hexilee — COMMENTED** [review]
Rest LGTM.

File: src/hookfs/mod.rs (hunk 1 above)
Comment (AsterNighT) [reply src/hookfs/mod.rs]: I wonder if there is a "rustuses" like "goimports" to sort imports. I'll look into it.

File: src/jsonrpc.rs (hunk 2 above)
Comment (AsterNighT) [reply src/jsonrpc.rs:88]: There can be poly-fill like `async_trait` and the lib itself seems to run on some multithread environment. I will look into it.

File: src/hookfs/mod.rs (hunk 1 above)
Comment (AsterNighT) [reply src/hookfs/mod.rs]: Good, there's a nightly feature in rustfmt for this

File: src/jsonrpc.rs (hunk 2 above)
Comment (AsterNighT) [reply src/jsonrpc.rs:88]: Oops, async trait is already being used in toda. That's awsome.

Comment (AsterNighT) [reply src/jsonrpc.rs:88]: https://github.com/paritytech/jsonrpc/issues/571
It seems that they are still on the way. And I see no urgent need on using an async structure at once. Maybe we can put it aside and wait for the issue.

**Comment by AsterNighT** [comment]
The way of exiting on rpc call is inconsistent when killing toda. Reverted to exit on signal. PTAL @YangKeao @Hexilee

**Review by Hexilee — APPROVED** [review]
LGTM

**Comment by AsterNighT** [comment]
@YangKeao

**Pull request merged**